### DISCUSSION TITLE
Team A - Describe components

### DISCUSSION THREAD

**Discussion by eunjungchoi** [lablup/contribution-academy-2021#81] [Show and tell]
A팀 팀장님인 @VSFe (김병철) 님이 조금 바쁘신 관계로 
팀원인 제가 쓰레드를 생성하였습니다. 

**Webserver**: 웹 애플리케이션을 호스팅하는  데몬으로서,  웹 클라이언트에서 aiohttp request가 오면 웹앱에 요청을 보내고 응답을 처리한다. (webui를 SPA로 제공하고 API 요청을 대리 함)
**Manager** : API 게이트웨이로서, 에이전트 및 컴퓨팅 커널을  추상화해서 관리하며 요청에 따라 해당 에이전트 및 세션으로 라우팅한다.
**Agent** : 워커의 상태 및 사용가능한 리소스 슬롯을 매니저에게 보고하고, 커널 컨테이너를 관리하고 제어하는 소형 데몬이다.

**Comment by VSFe** [comment]
**common** : 다른 컴포넌트에서 공통적으로 활용하는 라이브러리 컴포넌트
**client-py** : Python 환경에서 manager와 통신할 수 있도록 API Wrapper 관련 클래스를 제공하는 라이브러리
**kernels** : Backend.ai 환경에 최적화 되어 필요에 따라 사용할 수 있는 docker image를 모아놓은 저장소
**storage-proxy** : agent가 파일 저장 및 접근을 할 수 있도록 가상 폴더를 생성하고, 관리하는 컴포넌트
**webui** :  GUI 환경을 제공하기 위한 웹컴포넌트 구현체로, 일반적인 SPA 및 Electron 기반의 데스크톱 앱 관련 환경을 제공함.
